test-data/unit/check-warnings.test

```diff
@@ -145,8 +145,29 @@ main:4: warning: Returning Any from function declared to return "int"
 
 [case testReturnAnyForNotImplementedInBinaryMagicMethods]
 # flags: --warn-return-any
+from typing import Any
 class A:
-    def __eq__(self, other: object) -> bool: return NotImplemented
+    def __init__(self, x: Any) -> None:
+      self.x = x
+    def __eq__(self, other: Any) -> bool:
+      if self.__class__ is other.__class__:
+        return bool(self.x == other.x)
+      return NotImplemented
+    def __lt__(self, other: Any) -> bool: return NotImplemented
+    def __gt__(self, other: Any) -> bool: return NotImplemented
+    def __le__(self, other: Any) -> bool: return NotImplemented
+    def __ge__(self, other: Any) -> bool: return NotImplemented
+    def __ne__(self, other: Any) -> bool:
+      if self.__class__ is other.__class__:
+        return bool(self.x != other.x)
+      else:
+        # pylint will complain about this return not needing
+        # an "else", but want to check that Union[bool, NotImplemented]
+        # works as the inferred return typ
+        return NotImplemented
+    # TODO: If the programmer specifies "-> Union[bool, type(NotImplemented)]",
+    #       mypy suggests changing this to "-> Union[bool, type[NotImplemented]]",
```

Comment:
Does `Type[NotImplemented]` work? Maybe we should suggest that.

Reply:
```
>>> from typing import *
>>> Type[NotImplemented]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.5/typing.py", line 996, in __getitem__
    params = tuple(_type_check(p, msg) for p in params)
  File "/usr/lib/python3.5/typing.py", line 996, in <genexpr>
    params = tuple(_type_check(p, msg) for p in params)
  File "/usr/lib/python3.5/typing.py", line 312, in _type_check
    raise TypeError(msg + " Got %.100r." % (arg,))
TypeError: Parameters to generic types must be types. Got NotImplemented.
```

Reply:
That's a runtime error, and there's nothing you can do about it. The return type should just be `-> bool`.